plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/AbstractSQLDialect.java

```diff
@@ -822,25 +823,35 @@ public void generateStoredProcedureCall(StringBuilder sql, DBSProcedure proc, Co
         if (!inParameters.isEmpty()) {
             boolean first = true;
             for (DBSProcedureParameter parameter : inParameters) {
+                String typeName = parameter.getParameterType().getFullTypeName();
                 switch (parameter.getParameterKind()) {
                     case IN:
                         if (!first) {
-                            sql.append(",");
+                            sql.append(", ");
+                        }
+                        if (castParams) {
```

Comment:
You can put PostgreSQL-specific cast syntax to abstract SQL dialect. It has to be in PostgreDialect

Reply:
Sorry, I dont agree. This is common syntax, not only for postgres. It brings more help to postgres, but applicable as 'common sql syntax'

Reply:
According to SQL-2003 grammar cast is defined as `CAST (<cast operand>AS <cast target>)`.
So, it's not something PostgreSQL-specific unless we want to add this feature only for PostgreSQL for now.

Reply:
The thing is I'm not sure about is `:` before parameter name.

Reply:
Why? This works fine.

Yes, you are right. The PG specific cast is "<cast operand>::<cast target>" and I intentionally did not use it (also :: conflicts with :param).

Reply:
I'm not sure, that :param-name is supported in all databases

Reply:
I left this design as it was originally. There are 2 variations in code - ":param" and question mark.

Reply:
The `:param` is DBeaver-specific syntax.

DBeaver will prompt the user to enter values for these parameters before executing the query.

Reply:
@serge-rider , there is harcoded : (colon) in your existing code, so I left this logic unchanged. If it shoud be updated, couuld you give me an example how to use SQL_NAMED_PARAMETERS_PREFIX ?

Reply:
@plotn like this
```
DBPPreferenceStore prefStore;
DBPDataSource dataSource = proc.getDataSource();
if (dataSource != null) {
    prefStore = dataSource.getContainer().getPreferenceStore();
} else {
    prefStore = DBWorkbench.getPlatform().getPreferenceStore();
}
String namedParameterPrefix = prefStore.getString(ModelPreferences.SQL_NAMED_PARAMETERS_PREFIX);
```

Reply:
Okay! it works